src/Tokenizer/Transformer/ReturnRefTransformer.php

```diff
@@ -47,7 +47,7 @@ public function process(Tokens $tokens, Token $token, int $index): void
         }
 
         if (
-            $token->equals('&')
+            $token->isAmpersand()
```

Comment:
you should check PHP 7 `'&'` and PHP 8.1 `T_AMPERSAND_NOT_FOLLOWED_BY_VAR_OR_VARARG`
why checking `T_AMPERSAND_FOLLOWED_BY_VAR_OR_VARARG` at all here?

similar concern for other places.
those tokens have different context from 8.1, we should check the one we are interested in, not blindly checking both of them.